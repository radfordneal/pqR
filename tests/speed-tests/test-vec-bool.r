source("time.r")
 
a <- seq(1,3,length=1000)
b <- seq(2,4,length=1000)
as <- as.character(as.integer(100*a))
bs <- as.character(as.integer(100*b))
s <- seq(1,4,length=4)
n <- 200000

test.name <- "vec-bool.<="
f <- test.cmp (function () { for (i in 1:n) r <- 3<=a; r})
sys.time(r <- f()); print(sum(r))
f <- test.cmp (function () { for (i in 1:n) r <- a<=3; r})
sys.time(r <- f()); print(sum(r))
f <- test.cmp (function () { for (i in 1:n) r <- a<=b; r})
sys.time(r <- f()); print(sum(r))
f <- test.cmp (function () { for (i in 1:n) r <- a<=s; r})
sys.time(r <- f()); print(sum(r))
f <- test.cmp (function () { for (i in 1:(n/10)) r <- as<=bs; r})
sys.time(r <- f()); print(sum(as.integer(r)))

test.name <- "vec-bool.>="
f <- test.cmp (function () { for (i in 1:n) r <- 3>=a; r})
sys.time(r <- f()); print(sum(r))
f <- test.cmp (function () { for (i in 1:n) r <- a>=3; r})
sys.time(r <- f()); print(sum(r))
f <- test.cmp (function () { for (i in 1:n) r <- a>=b; r})
sys.time(r <- f()); print(sum(r))
f <- test.cmp (function () { for (i in 1:n) r <- a>=s; r})
sys.time(r <- f()); print(sum(r))
f <- test.cmp (function () { for (i in 1:(n/10)) r <- as>=bs; r})
sys.time(r <- f()); print(sum(as.integer(r)))

test.name <- "vec-bool.<"
f <- test.cmp (function () { for (i in 1:n) r <- 3<a; r})
sys.time(r <- f()); print(sum(r))
f <- test.cmp (function () { for (i in 1:n) r <- a<3; r})
sys.time(r <- f()); print(sum(r))
f <- test.cmp (function () { for (i in 1:n) r <- a<b; r})
sys.time(r <- f()); print(sum(r))
f <- test.cmp (function () { for (i in 1:n) r <- a<s; r})
sys.time(r <- f()); print(sum(r))
f <- test.cmp (function () { for (i in 1:(n/10)) r <- as<bs; r})
sys.time(r <- f()); print(sum(as.integer(r)))

test.name <- "vec-bool.>"
f <- test.cmp (function () { for (i in 1:n) r <- 3>a; r})
sys.time(r <- f()); print(sum(r))
f <- test.cmp (function () { for (i in 1:n) r <- a>3; r})
sys.time(r <- f()); print(sum(r))
f <- test.cmp (function () { for (i in 1:n) r <- a>b; r})
sys.time(r <- f()); print(sum(r))
f <- test.cmp (function () { for (i in 1:n) r <- a>s; r})
sys.time(r <- f()); print(sum(r))
f <- test.cmp (function () { for (i in 1:(n/10)) r <- as>bs; r})
sys.time(r <- f()); print(sum(as.integer(r)))

test.name <- "vec-bool.=="
f <- test.cmp (function () { for (i in 1:n) r <- 3==a; r})
sys.time(r <- f()); print(sum(r))
f <- test.cmp (function () { for (i in 1:n) r <- a==3; r})
sys.time(r <- f()); print(sum(r))
f <- test.cmp (function () { for (i in 1:n) r <- a==b; r})
sys.time(r <- f()); print(sum(r))
f <- test.cmp (function () { for (i in 1:n) r <- a==s; r})
sys.time(r <- f()); print(sum(r))
f <- test.cmp (function () { for (i in 1:(n/2)) r <- as==bs; r})
sys.time(r <- f()); print(sum(as.integer(r)))

test.name <- "vec-bool.!="
f <- test.cmp (function () { for (i in 1:n) r <- 3!=a; r})
sys.time(r <- f()); print(sum(r))
f <- test.cmp (function () { for (i in 1:n) r <- a!=3; r})
sys.time(r <- f()); print(sum(r))
f <- test.cmp (function () { for (i in 1:n) r <- a!=b; r})
sys.time(r <- f()); print(sum(r))
f <- test.cmp (function () { for (i in 1:n) r <- a!=s; r})
sys.time(r <- f()); print(sum(r))
f <- test.cmp (function () { for (i in 1:(n/2)) r <- as!=bs; r})
sys.time(r <- f()); print(sum(as.integer(r)))

u <- a>1.5
v <- b<3.2
w <- s==2
n <- 200000

test.name <- "vec-bool.&"
f <- test.cmp (function () { for (i in 1:n) r <- u&v; r})
sys.time(r <- f()); print(sum(r))
f <- test.cmp (function () { for (i in 1:n) r <- u&w; r})
sys.time(r <- f()); print(sum(r))
f <- test.cmp (function () { for (i in 1:n) r <- w&v; r})
sys.time(r <- f()); print(sum(r))

test.name <- "vec-bool.|"
f <- test.cmp (function () { for (i in 1:n) r <- u|v; r})
sys.time(r <- f()); print(sum(r))
f <- test.cmp (function () { for (i in 1:n) r <- u|w; r})
sys.time(r <- f()); print(sum(r))
f <- test.cmp (function () { for (i in 1:n) r <- w|v; r})
sys.time(r <- f()); print(sum(r))
