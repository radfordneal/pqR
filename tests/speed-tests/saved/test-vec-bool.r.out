
---- vec-bool.<= 

Time:
[1] 1
Time:
[1] 1000
Time:
[1] 1000
Time:
[1] 626
Time:
[1] 1000

---- vec-bool.>= 

Time:
[1] 1000
Time:
[1] 1
Time:
[1] 0
Time:
[1] 375
Time:
[1] 0

---- vec-bool.< 

Time:
[1] 0
Time:
[1] 999
Time:
[1] 1000
Time:
[1] 625
Time:
[1] 1000

---- vec-bool.> 

Time:
[1] 999
Time:
[1] 0
Time:
[1] 0
Time:
[1] 374
Time:
[1] 0

---- vec-bool.== 

Time:
[1] 1
Time:
[1] 1
Time:
[1] 0
Time:
[1] 1
Time:
[1] 0

---- vec-bool.!= 

Time:
[1] 999
Time:
[1] 999
Time:
[1] 1000
Time:
[1] 999
Time:
[1] 1000

---- vec-bool.& 

Time:
[1] 350
Time:
[1] 187
Time:
[1] 150

---- vec-bool.| 

Time:
[1] 1000
Time:
[1] 813
Time:
[1] 700
